qualtran/bloqs/arithmetic/subtraction_test.py

```diff
@@ -21,15 +21,20 @@
 from qualtran.resource_counting.generalizers import ignore_split_join
 
 
-def test_subtract_bloq_decomposition():
-    gate = Subtract(QInt(3), QInt(5))
+@pytest.mark.parametrize(
+    ['a_bits', 'b_bits'], [(a, b) for a in range(1, 6) for b in range(a, 6) if a + b <= 10]
+)
+def test_subtract_bloq_decomposition(a_bits, b_bits):
```

Comment:
can you add a test that uses the classical simulation protocol

```python
cbloq = subtract.decompose_bloq()

for in_vals:
  ref = subtract.call_classically(in_vals)
  comp = cbloq.call_classically(in_vals)
  assert ref == comp
```

Reply:
done